### PR TITLE
[Port to C++] Blob Detection

### PR DESCRIPTION
**Pull request by Khushi-Balia**

### CODE REVIEW

**Review by amanchhaparia — CHANGES_REQUESTED** [review]
@Khushi-Balia, requesting the below changes.
- Add table of content in the Readme
- Rename readme.md to `README.md`
- Refer convolution and filtering code structure ie have a `src`, `include` folders.
- Refer convolution and filtering code files and follow the similar fashion for adding the comments for the functions
- Remove the unnecessary `blob` file and not required files from the folder.

**Comment by Khushi-Balia** [comment]
Made the required changes.

File: 4_cv_basics/7_blob_detection/README.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
+# Blob Detection
+
+BLOB stands for Binary Large Object
+
+Informally a blob is a region of an image in which some properties like intensity or color are approximately constant.
+
+## Table of Contents
+
+- [Understanding the code](#understanding-the-code)
+- [Detecting the blob](#detecting-the-blob)
```
Comment (amanchhaparia) [inline 4_cv_basics/7_blob_detection/README.md:10]: Can be more informative if the contents in the Readme can be divided into the different topic sections similar to [this](https://github.com/SRA-VJTI/Pixels_Seminar/blob/main/3_cv_basics/6_blob_detection.ipynb)

File: 4_cv_basics/7_blob_detection/include/blob_detection.hpp [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+#include <opencv2/opencv.hpp>
+
+using namespace std;
+using namespace cv;
+
+double median(vector<double> vec);
+
+tuple<double, double, double> getMedianPixelValues(Mat img);
+
+#endif
```
Comment (amanchhaparia) [inline 4_cv_basics/7_blob_detection/include/blob_detection.hpp:13]: Add a new line

File: 4_cv_basics/7_blob_detection/main.cpp [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        if (waitKey(10) == 'x')
+        {
+            destroyAllWindows();
+            video.release();
+            break;
+        }
+    }
+
+    return 0;
+}
```
Comment (amanchhaparia) [inline 4_cv_basics/7_blob_detection/main.cpp:94]: Add a newline

**Review by amanchhaparia — CHANGES_REQUESTED** [review]

**Comment by Khushi-Balia** [comment]
Updated the Readme and made the changes.

**Comment by aPR0T0** [comment]
@Khushi-Balia when we run the code for blob detection the window that pops-up is nearly my screen size, so just make it resizable and keep initial size of window smaller for better handling

**Comment by amanchhaparia** [comment]
@Khushi-Balia, in the Readme content there are certain part which referred to python functions. Kindly update them. Also move `main.cpp` to `src`.

**Comment by Khushi-Balia** [comment]
Made the windows resizable, updated the readme and file structure.

**Comment by Khushi-Balia** [comment]
Resolved the error, made the necessary changes in the Makefile.

**Comment by amanchhaparia** [comment]
@Khushi-Balia, resize is not working. I guess you need to first define the window then resize it.

**Comment by Khushi-Balia** [comment]
Changed the window size and updated the readme.

**Comment by amanchhaparia** [comment]
@Khushi-Balia, update the PR according to the new file structure.

**Review by amanchhaparia — APPROVED** [review]

**Pull request merged**